core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -122,6 +123,28 @@ class SnappyParser(session: SnappySession)
     }
   }
 
+  private final def updatePerTableQueryHint(tableIdent: TableIdentifier,
+      optAlias: Option[String]) = {
+    val indexHint = queryHints.remove(QueryHint.Index.toString)
+    if (indexHint.nonEmpty) {
+      val table = optAlias match {
+        case Some(alias) => alias
+        case _ => tableIdent.unquotedString
+      }
+
+      queryHints.put(QueryHint.Index.toString + table, indexHint.get)
```

Comment:
queryHints gets cleared across multiple queries. If you mean multiple tables, it is per tableIdentifier.

select * from t1 /*++index(dd)*/ , t2 /*++index(d3)*/ where ....